internal/k8s/validation.go

```diff
@@ -572,6 +583,22 @@ func validateSnippetsAnnotation(context *annotationValidationContext) field.Erro
 	return allErrs
 }
 
+func validateSecretNameAnnotation(context *annotationValidationContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if err := validateIsValidSecretName(context.value); err != nil {
```

Comment:
Suggestion: There is a kubernetes apimachinery validation function that does the same thing as this new function: https://pkg.go.dev/k8s.io/apimachinery@v0.23.6/pkg/util/validation#IsDNS1123Subdomain